src/server/lesson-controller.js

```diff
@@ -37,16 +38,22 @@ class LessonController {
   async handleGetLessonPage(req, res) {
     const { user } = req;
     const { lessonId } = req.params;
+    const { view, templateLessonId } = req.query;
     const routeWildcardValue = urls.removeLeadingSlash(req.params['0']);
 
     const lesson = await this.lessonService.getLessonById(lessonId);
+    const templateLesson = templateLessonId ? await this.lessonService.getLessonById(templateLessonId) : null;
 
-    if (!lesson) {
+    if (!lesson || (templateLessonId && !templateLesson)) {
       throw new NotFound();
     }
 
+    if (templateLessonId && lesson.roomId !== templateLesson.roomId) {
```

Comment:
Tiny thing: at this point would probably make more sense to use `templateLesson` instead of `templateLessonId` in this condition 🤔

Reply:
True, thanks, changed it to `templateLesson`